### ISSUE TITLE
Expand query system to allow getting selected attribute only

### ISSUE THREAD

**Issue by dominiklohmann** [koekeishiya/yabai#18]
I'm trying to re-implement `chunkc tiling::query --monitor-for-desktop <id>` and have done so using text replacements:

```sh
yabai -m query --spaces --space 2 | sed -n -e 's/^.*"monitor":\(.*\),/\1/p'
```

The syntax I'm proposing as a replacement is

```sh
yabai -m query --spaces --space 2 monitor
```

which should return just the monitor id of space 2 (and a newline at the end of the output).

**Comment by acmcelwee** [comment]
FWIW, [jq](https://stedolan.github.io/jq/manual/) is a much simpler way to process the existing query outputs than sed. Example: 
```
$ yabai -m query --spaces --space 2 | jq .monitor
1
```

**Comment by dominiklohmann** [comment]
Didn't know `jq`. Lovely little tool. Thanks!

**Comment by koekeishiya** [comment]
Just for info, I'm going to rename the "monitor" key to "display" to remain consistent with the rest of the program. Not sure how I made that mistake here..

**Comment by koekeishiya** [comment]
Oh and you could also use `yabai -m query --displays --space | jq .index` to get the display of the active space.

**Comment by dominiklohmann** [comment]
On a similar note, I noticed that I can do stuff like this with `jq`:

```
# get all known open windows in a "app: window title" format
yabai -m query --windows | jq '.[] | .app + ": " + .title'

# get the space count of all displays
yabai -m query --displays | jq '.[].spaces'
```

Really amazing tool, makes this stuff so easy once you grok the syntax.

**Comment by koekeishiya** [comment]
On a related note - has anyone looked into how Spotlight works and if it is possible to create custom searchable text and commands for it? As focusing the proper window has now been resolved using the scripting addition, and with the information provided through the query system in yabai I think it would be handy to create a fuzzy-seach mechanism for focusing windows - this could let you filter windows visible on the current space, current display(?) or all known windows, or whatever.

**Comment by dominiklohmann** [comment]
There's [Flashlight](https://flashlight.nateparrott.com/), which basically is an extension to Spotlight, but I'm not sure if it still works. It's never been running smoothly for me, although I've only tried it after it first came out. I since switched back to Alfred—they recently released v4—, which has workflows that are very easy to configure. It's basically Spotlight on steroids and much better in every regard, especially responsiveness.
